scrimage-core/src/main/java/com/sksamuel/scrimage/nio/internal/GifSequenceReader.java

```diff
@@ -760,7 +778,9 @@ protected int readShort() {
     * Resets frame state for reading next image.
     */
    protected void resetFrame() {
-      lastDispose = dispose;
+      if (dispose == 1) {
+         this.frameIndexWithLastDoNotDispose = frameCount-1;
```

Comment:
Instead of capturing and maintaining the last dispose method, we want to instead capture the last frame to declare a doNotDispose method, that way we can properly set the background image for the 'restoreToPrevious' method. Figure 23-1 from https://docstore.mik.ua/orelly/web2/wdesign/ch23_05.htm was extremely helpful in understanding that particular method.